crates/datasources/src/sqlite/async_sqlite.rs

```diff
@@ -0,0 +1,232 @@
+use std::path::PathBuf;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use datafusion::arrow::datatypes::DataType;
+use futures::Stream;
+use rusqlite::types::Value;
+use rusqlite::{Connection, Params};
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::error::TrySendError;
+use tokio::task::JoinHandle;
+
+use crate::sqlite::errors::{Result, SqliteError};
+
+#[derive(Clone, Debug)]
+pub struct SqliteAsyncClient {
+    path: PathBuf,
+}
+
+impl SqliteAsyncClient {
+    pub fn new(path: PathBuf) -> Self {
+        Self { path }
+    }
+
+    pub fn execute(&self, s: impl Into<String>, p: impl Params) -> Result<usize> {
+        let s = s.into();
+        let path = self.path.clone();
+
+        tokio::task::block_in_place(move || {
+            let conn = Connection::open(path)?;
+            let n = conn.execute(&s, p)?;
+            Ok(n)
+        })
+    }
+
+    pub fn query(
+        &self,
+        s: impl Into<String>,
+        p: impl Params + Send + 'static,
+    ) -> SqliteQueryStream {
+        let s = s.into();
+
+        let (req_tx, mut req_rx) = mpsc::channel(1);
+        let (resp_tx, resp_rx) = mpsc::channel(1);
+
+        let path = self.path.clone();
+        let handle: JoinHandle<Result<()>> = tokio::task::spawn_blocking(move || {
```

Comment:
This should probably just be `spawn` to allow for non-blocking recvs on the channel. Otherwise this will literally block a thread.